libgnucash/app-utils/options.scm

```diff
@@ -1688,33 +1688,46 @@
-                        "Use Full Other Account Name?" "Use Full Other Account Name"
-                        "Void Transactions?" "Void Transactions"
-                        ))
+                                        "Accounts to include" (cons #f "Accounts")
+                                        "Exclude transactions between selected accounts?" (cons #f "Exclude transactions between selected accounts")
+                                        "Filter Accounts" (cons #f "Filter By...")
+                                        "Flatten list to depth limit?" (cons #f "Flatten list to depth limit")
+                                        "From" (cons #f "Start Date")
+                                        "Report Accounts" (cons #f "Accounts")
+                                        "Report Currency" (cons #f "Report's currency")
+                                        "Show Account Code?" (cons #f "Show Account Code")
+                                        "Show Full Account Name?" (cons #f "Show Full Account Name")
+                                        "Show Multi-currency Totals?" (cons #f "Show Multi-currency Totals")
+                                        "Show zero balance items?" (cons #f "Show zero balance items")
+                                        "Sign Reverses?" (cons #f "Sign Reverses")
+                                        "To" (cons #f "End Date")
+                                        "Use Full Account Name?" (cons #f "Use Full Account Name")
+                                        "Use Full Other Account Name?" (cons #f "Use Full Other Account Name")
+                                        "Void Transactions?" (cons "Filter" "Void Transactions")
+                                        "Void Transactions" (cons "Filter" "Void Transactions")
+                                        "Account Substring" (cons "Filter" "Account Matcher")
+                                        ))
                        (name-match (member name new-names-list)))
 
-                      (if name-match 
-                        (let ((new-name (cadr name-match)))
-                             (lookup-option section new-name))
-                        #f))))
-          #f)))
+                  (and name-match
+                       (let ((new-section (car (cadr name-match)))
+                             (new-name (cdr (cadr name-match))))
+                         ;; compare if new-section name exists.
+                         (if new-section
+                             ;; if so, if it's different to current section name
+                             ;; then try new section name
+                             (and (not (string=? new-section section))
+                                  (lookup-option new-section new-name))
+                             ;; else reuse section-name with new-name
+                             (lookup-option section new-name))))))))
+      #f)))
```

Comment:
There is a closing parenthesis mismatch in here somewhere. When I pull this PR I get a build error (using guile 2.2):
```
options.scm:1669:0: no expressions in body in form (lambda () (define option-hash (make-hash-table 23)) (define options-changed #f) (define changed-hash (make-hash-table 23)) (define callback-hash (make-hash-table 23)) (define last-callback-id 0) (define (lookup-option section name) (let ((section-hash (hash-ref option-hash section))) (if section-hash (let ((option-hash (hash-ref section-hash name))) (if option-hash option-hash (let* ((new-names-list (list "Accounts to include" (cons #f "Accounts") "Exclude transactions between selected accounts?" (cons #f "Exclude transactions between selected accounts") "Filter Accounts" (cons #f "Filter By...") "Flatten list to depth limit?" (cons #f "Flatten list to depth limit") "From" (cons #f "Start Date") "Report Accounts" (cons #f "Accounts") "Report Currency" (cons #f "Report's currency") "Show Account Code?" (cons #f "Show Account Code") "Show Full Account Name?" (cons #f "Show Full Account Name") "Show Multi-currency Totals?" (cons #f "Show Multi-currency Totals") "Show zero balance items?" (cons #f "Show zero balance items") "Sign Reverses?" (cons #f "Sign Reverses") "To" (cons #f "End Date") "Use Full Account Name?" (cons #f "Use Full Account Name") "Use Full Other Account Name?" (cons #f "Use Full Other Account Name") "Void Transactions?" (cons "Filter" "Void Transactions") "Void Transactions" (cons "Filter" "Void Transactions"))) (name-match (member name new-names-list))) (and name-match (let ((new-section (car (cadr name-match))) (new-name (cdr (cadr name-match)))) (if new-section (and (not (string=? new-section section)) (lookup-option new-section new-name)) (lookup-option section new-name)))))))) #f)))
```

Looking at options.scm, there's an unmatched closing parenthesis 1899, but the issue starts somewere in the code above.

Also while studying this, it occurred to me this code will not handle renaming a complete section. It will only handle moving an option from one (still existing) section to another and renaming the option.

We haven't ever renamed a complete section yet, but it may be worth adding a comment to show at least we considered it and are aware this is currently not supported.

Reply:
Pushed 1 change, will do further TR work tonight.